src/instantsearch/instantsearch.ts

```diff
@@ -158,13 +159,15 @@ export type InstantSearchConfig = {
   appId?: string;
   apiKey?: string;
   indexName: string;
+  options?: ClientOptions;
```

Comment:
Why do you need to add this prop? `options` can be directly passed to the `searchClient` prop (`appId` and `apiKey` are deprecated and not supported from v3):

```js
const searchClient = algoliasearch('appId', 'apiKey', options);
```

Reply:
this seems to be for server side rendering, where we indeed don't expose the options. It would make more sense for these options to go _only_ to createSSRClient. 

Which options do you use?

Reply:
First off all, thank you for reviewing my code, and sorry since my knowledge of TypeScript and frontend is limited, made some mess.

I wan't to set up `hosts.read` and `hosts.write` to setup my proxy host, which will track product ids returned for a set of filters and etc.

Additionally, we will use more than one Algolia cluster so my `proxy` would balance between them without client modifications.